script.js

```diff
@@ -0,0 +1,162 @@
+		$("#message").show();
+	}
+}
+
+function nodePermission(event){
+	var input = event.target;
+	var currentValue = input.value;
+	var inputId = input.id;
+	
+	arrayService.updateNode(input);
+	if (arrayService.isNodeValid(+input.id.slice(4))){
+		$("#"+inputId).removeClass("invalidNode");	
+	}
+	else{
+		$("#"+inputId).addClass("invalidNode");
+	}
+	var nextStepBtn = document.getElementById("nextStepBtn");
+	if (arrayService.isArrayValid()){
+		enableButton(nextStepBtn);
+	} else{
+		disableButton(nextStepBtn);
+	}
+}
+
+function reloadPage(){
+	var node0 = document.getElementById("node0");
+	var node1 = document.getElementById("node1");
+	node0.value = "";
+	node1.value = "";
+	node0.disabled = false;
+	node1.disabled = false;
+	
+	unlightNode(node0.id);
+	unlightNode(node1.id);
+	unlightNodeAsStable(node0.id);
+	unlightNodeAsStable(node1.id);
+	$("#node0").removeClass("invalidNode");
+	$("#node0").removeClass("sortNode");
+	$("#node1").removeClass("invalidNode");
+	$("#node1").removeClass("sortNode");
+	$("#message").hide();
+	
+	var newNodeBtn = document.getElementById("newNodeBtn");
+	enableButton(newNodeBtn);
+	var nextStepBtn = document.getElementById("nextStepBtn");
+	disableButton(nextStepBtn);
+	
+	arrayService.initArray([node0, node1]);
+}
+
+function onLoad(){
+	reloadPage();
+}
+
+function onReset() {
+	var nodeDiv = document.getElementById("nodeDiv");
+	for (var i = arrayService.getLength()-1; i > 1; i--){
+		var node = document.getElementById("node".concat(i));
+		nodeDiv.removeChild(node);
+	}
+	$("#nextStepBtn").show();	
+	reloadPage();
+}
+
+function startSort(){
+	toSortMode();
+	var stepResult = arrayService.sortStep();
+	
+	if (stepResult.newStableNodeIds[1] === "node0"){
+		$("#nextStepBtn").hide();
+	}
+	
+	stepResult.prevNodeIds.forEach(function(nodeId){
+		unlightNode(nodeId);
+	});
+	
+	stepResult.activeNodeIds.forEach(function(nodeId){
+		highlightNode(nodeId);
+	});
+	
+	stepResult.newStableNodeIds.forEach(function(nodeId){
+		highlightNodeAsStable(nodeId);
+	});
+}
+
+function toSortMode() {
+	for (var i = 0; i < arrayService.getLength(); i++){
+		var node = $("#node".concat(i));
+		node[0].disabled = true;
+		node.addClass("sortNode");
+	}
+	var newNodeBtn = document.getElementById("newNodeBtn");
+	disableButton(newNodeBtn);
+}
+
+function showMessage(){
+	var message = document.getElementById("message");
+	message.style.visibility = "visible";
+}
+
+function hideMessage(){
+	var message = document.getElementById("message");
+	message.style.visibility = "hidden";
+}
+
+function disableButton(btn){
+	btn.disabled = true;
+}
+
+function enableButton(btn){
+	btn.disabled = false;
+}
+
+function highlightNode(nodeId){
+	$("#"+nodeId).addClass("activeNode");
+}
+
+function highlightNodeAsStable(nodeId){
+	$("#"+nodeId).addClass("stableNode");
+}
+
+function unlightNodeAsStable(nodeId){
+	$("#"+nodeId).removeClass("stableNode");
+}
+
+function unlightNode(nodeId){
+	$("#"+nodeId).removeClass("activeNode");
+}
```

Comment:
The code is clear, the coding style is consistent for both .js files. Variables and functions have clear names which describes their functionality in code.